internal/http/client.go

```diff
@@ -0,0 +1,77 @@
+package http
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"os"
+)
+
+type Client struct {
+	client *http.Client
+}
+
+func NewHTTPClient() *Client {
+
+	tlsConfig, err := CreateTLSClientConfig()
+	if err != nil {
+		return nil
```

Comment:
It's good practice to propagate the error back up to the caller, letting it handle the error appropriately.